data-prepper-plugins/dynamodb-source/src/main/java/org/opensearch/dataprepper/plugins/source/dynamodb/converter/RecordConverter.java

```diff
@@ -91,12 +92,15 @@ public void addToBuffer(Map<String, Object> data, Map<String, Object> keys, long
             eventMetadata.setAttribute(PRIMARY_KEY_DOCUMENT_ID_METADATA_ATTRIBUTE, partitionKey);
         }
         bufferAccumulator.add(new Record<>(event));
+        if (acknowledgementSet != null) {
+            acknowledgementSet.add(event);
```

Comment:
I didn't fully understand how this works, but will this have any memory issue? I only see there is a add of event, will that be a clear of event somewhere else.

Reply:
This just adds an `eventHandle` to the acknowledgment set, and this takes up very small amount of memory. @kkondaka can explain more